core-rust/state-manager/src/store/historical_state.rs

```diff
@@ -259,6 +259,19 @@ impl<'s, R: ReadableRocks> VersionScopedSubstateDatabase<'s, StateManagerDatabas
     }
 }
 
+/// Validates the requested historical state version and returns it (if present), or returns the
+/// current state version otherwise.
+///
+/// Implementation note:
+/// We re-use the validation performed by the [`VersionScopedSubstateDatabase::new()`] (which means
+/// throwing away a constructed instance).
+pub fn resolve_effective_state_version<R: ReadableRocks>(
```

Comment:
I'm not a big fan of these methods (this one and `VersionScopedSubstateDatabase::new`) taking an optional state version. This seems more of a handler concern. Wdyt about moving it to a helper method somewhere in `node-http-apis` (and only have two separate methods in `store` e.g. `new_latest()` and `new_at_state_version(non_optional_state_version)`). Not a big deal though :)

Reply:
so this particular method gets replaced by https://github.com/radixdlt/babylon-node/pull/889/files#diff-c85ce3e1172449a43cccc5b24780b578fc136f1afc88826fcff465ac5b475a56R235 - does it look "less ugly"?

I agree that e.g. routing to current vs historical `store` construction is the handler's responsibility, but it is kinda related to determining whether it is an available past version, etc (which reaches out to DB). So it is just convenient to do near the DB...